src/Microsoft.Identity.Web/TokenCacheProviders/Distributed/MsalDistributedTokenCacheAdapter.Log.cs

```diff
@@ -0,0 +1,269 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Identity.Web.TokenCacheProviders.Distributed
+{
+    /// <summary>
+    /// LoggingMessage class for MsalDistributedTokenCacheAdapter.
+    /// </summary>
+    public partial class MsalDistributedTokenCacheAdapter
+    {
+        /// <summary>
+        /// LoggingMessage class for MsalDistributedTokenCacheAdapter.
+        /// </summary>
+        private static class Log
+        {
+            private static readonly Action<ILogger, string, string, string, int, bool, Exception?> s_l2CacheState =
+                LoggerMessage.Define<string, string, string, int, bool>(
+                    LogLevel.Debug,
+                    new EventId(2, "DistributedCacheState"),
```

Comment:
>2 [](start = 32, length = 1)

All event id's are 2 (or 1 or 3) depending on the class, not sure how maintanable this is across multiple files, it requies search to find all existing used event id's. Perhaps separate class with this definition. (still trying to understand if it's expected to be unique per type or rather unique per message per type).

Reply:
addressed [here](https://github.com/AzureAD/microsoft-identity-web/pull/1108) and started discussion in teams.